201-virtual-wan-with-all-gateways/azuredeploy.json

```diff
@@ -0,0 +1,514 @@
+      "defaultValue": "MIIC5zCCAc+gAwIBAgIQWwgCEU04Eq5J9JXXT8LdwDANBgkqhkiG9w0BAQsFADAWMRQwEgYDVQQDDAtQMlNSb290Q2VydDAeFw0yMDEyMDUwODU2NTlaFw0yMTEyMDUwOTE2NTlaMBYxFDASBgNVBAMMC1AyU1Jvb3RDZXJ0MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA9XWoyfTT830otnzxejlNb2ZjP7DkxpI7e9k/vqJDt64HtugMa/4uTNQ7tDjU5rVSizKdsI40tiFH4dkdkoZefDLHpayyZrza8KzY3UUqSdoR1smHWAn+ojHVL3Gs4oTUlyRFbWLlBiw+Gw6c42GgGBQ8Alk4PXiVz3YDtre1e/W9cHcNdWCloL2hlx+EpkXhSxC4LaXPgrSMNcgV/o9LTvUwA8Bbk+3CfPd37vI5YElrHav9+pp4GOh73cMG/y8WGXnbmq8/FrwrJK7aQaYRDCucCGogZ6iMZU2/QLAAJrUld2j1jV9dMyOstjkmPaWjd/4Vi0A5sbCszWp57BWs7QIDAQABozEwLzAOBgNVHQ8BAf8EBAMCAgQwHQYDVR0OBBYEFLwzwL4MTZ4l+glwCKhA1xDhedi0MA0GCSqGSIb3DQEBCwUAA4IBAQBUhnXsAYXBwRmrL6fse7NxCjQvUZ6a49pyNGDw58hs+euQqkMH0YbVOYRkgdlR/Vi4zDg2nV/3+PCPXxdKWaa9c1eanIq/A1stjsJWo0LpGoGc1pvVQD6ufsGiXXAD4i7WFfXbaj2zGY/Jyc4ictpTzD1aaonIU5ku8Bi9a4HLt0RDzX4JnxuJE3052UojRI5l+scKbiZ1n/G5IJewecvkxt3ZeDrJfDqa6nvqwsijGH4QKkZ8GIxbhA3XIPo7tAL0vVdIwfKsUYVuVxNARfNgqoWTXFIkqHb7cOAI2mZ5SWs+fCt5UgKNdQT4lKMhfDCcBTuNJkp67Rlt4DM2igvp"
+    }
+
+  },
+  "functions": [],
+  "variables": {
+    "vwan_cfg": {
+      "type": "Standard"
+    },
+    "virtual_hub1_cfg": {
+      "name": "vhubvnet1",
+      "addressSpacePrefix": "192.168.0.0/24",
+      "Hub1_P2SvpnClientAddressPoolPrefix": "10.4.3.0/24"
+    },
+    "virtual_hub2_cfg": {
+      "name": "vhubvnet2",
+      "addressSpacePrefix": "192.168.1.0/24",
+      "Hub2_P2SvpnClientAddressPoolPrefix": "10.5.3.0/24"
+    },
+    "vnet1_cfg": {
+      "name": "VNET1",
+      "addressSpacePrefix": "10.1.0.0/24",
+      "subnetName": "subnet1",
+      "subnetPrefix": "10.1.0.0/24"
+    },
+    "vnet2_cfg": {
+      "name": "VNET2",
+      "addressSpacePrefix": "10.1.1.0/24",
+      "subnetName": "subnet1",
+      "subnetPrefix": "10.1.1.0/24"
+    },
+    "vnet3_cfg": {
+      "name": "VNET3",
+      "addressSpacePrefix": "10.2.0.0/24",
+      "subnetName": "subnet1",
+      "subnetPrefix": "10.2.0.0/24"
+    },
+    "vnet4_cfg": {
+      "name": "VNET4",
+      "addressSpacePrefix": "10.2.1.0/24",
+      "subnetName": "subnet1",
+      "subnetPrefix": "10.2.1.0/24"
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Network/virtualWans",
+      "apiVersion": "2020-05-01",
+      "name": "[parameters('vWANname')]",
+      "location": "[parameters('hub1_location')]",
+      "properties": {
+        "allowVnetToVnetTraffic": true,
+        "allowBranchToBranchTraffic": true,
+        "type": "[variables('vwan_cfg').type]"        
+      }
+    },
+    {
+      "type": "Microsoft.Network/virtualHubs",
+      "apiVersion": "2020-05-01",
+      "name": "[variables('virtual_hub1_cfg').name]",
+      "location": "[parameters('hub1_location')]",
+      "properties": {
+        "addressPrefix": "[variables('virtual_hub1_cfg').addressSpacePrefix]",
+        "virtualWan": {
+          "id": "[resourceId('Microsoft.Network/virtualWans', parameters('vWANname'))]"
+        }
+      },
+      "dependsOn": [
```

Comment:
See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties